pkg/libvmi/cloudinit.go

```diff
@@ -60,6 +62,29 @@ func WithCloudInitNoCloudNetworkData(data string) Option {
 	}
 }
 
+// WithCloudInitNoCloudEncodedNetworkData adds cloud-init no-cloud base64 encoded network data.
+func WithCloudInitNoCloudEncodedNetworkData(networkData string) Option {
+	return func(vmi *v1.VirtualMachineInstance) {
+		addDiskVolumeWithCloudInitNoCloud(vmi, cloudInitDiskName, v1.DiskBusVirtio)
+
+		volume := getVolume(vmi, cloudInitDiskName)
+		volume.CloudInitNoCloud.NetworkDataBase64 = base64.StdEncoding.EncodeToString([]byte(networkData))
+		volume.CloudInitNoCloud.NetworkData = ""
+	}
+}
+
+// WithCloudInitNoCloudNetworkDataSecret adds cloud-init no-cloud network data from secret.
+func WithCloudInitNoCloudNetworkDataSecret(secretName string) Option {
```

Comment:
I still prefer to pass here a secretRef: `secretRef *k8scorev1.LocalObjectReference`.
But it is not a blocker.

If you keep it like this, perhaps it is better to add `Name` at the end of the function name and mention it in the description.

Reply:
Done, added `Name` at the end.